### DISCUSSION TITLE
A JSON path cannot be set without a scalar filter?

### DISCUSSION THREAD

**Discussion by timeimp** [prisma/prisma#20357] [Q&A] [answered]
I have `findMany` query that I am running like so:

```ts
where: {
    actor: {
      path: ['identity', 'user'],
      equals: req.params.userId,
    }
}
```
`actor` is a `Json` (not `Json?`) field type, Postgres is the backing DB.

And when I attempt to run the request, Prisma fails with:

```
Input error. A JSON path cannot be set without a scalar filter.
```

`clientVersion: 5.0.0`

Can anyone say if I am doing anything wrong? I followed [this article](https://www.prisma.io/docs/concepts/components/prisma-client/working-with-fields/working-with-json-fields#filter-on-nested-object-property) but it seems to be broken?

As this query is only new, I have no way to validate it working under previous versions.

**Comment by ludralph** [comment]
Hi @timeimp 👋 

I was able to get the code in the [article](https://www.prisma.io/docs/concepts/components/prisma-client/working-with-fields/working-with-json-fields#filter-on-nested-object-property) working. Can you please share the structure of your `Json` object so I could reproduce it?

**Comment by timeimp** [comment]
Hey @ludralph!

The JSON field has the following format:

```ts
{
eventId: string,
identity: {
 user: string,
 service: string
}
```

Does it matter if these are formally defined inside an interface file at all?

Cheers,
timeimp

**Reply by ludralph** [reply]
Hi @timeimp

The JSON format you shared works for me. I can query it and get results.

> Does it matter if these are formally defined inside an interface file at all?

The JSON field does not necessarily need to be formally defined inside an interface file. It's important to clarify that Prisma doesn't use interface files for schema definitions. Prisma uses its own DSL (Domain-Specific Language) for schema definitions, typically stored in a .prisma file. However, if you are using TypeScript and want to define TypeScript interfaces for your data models, you can do that separately from Prisma. As long as you define the JSON field as a JSON scalar type in your Prisma schema, you can store JSON data with the format you provided without the need for formal interfaces within Prisma itself.

**Reply by timeimp** [reply]
Good to know!

To be sure, when you say

>As long as you define the JSON field as a JSON scalar type in your Prisma schema,

This is what you mean?

```prisma

model vHistory {
  id         Int      @id @default(autoincrement())
  guid       String   @unique @default(cuid())
  actor      Json
}
```

**Reply by ludralph** [reply]
Yes it is

**Comment by timeimp** [comment, marked as answer]
I worked it out! I had another query that was being called crashing the script and it had a `null` value.

Thanks @ludralph for your input!

To anyone else who finds this error in the future:

You need to make sure that the value of the `equals` part is not **NULL** but instead "DbNull" as defined here: https://www.prisma.io/docs/concepts/components/prisma-client/working-with-fields/working-with-json-fields#filtering-by-null-values

The error message doesn't seem too helpful until you realise that it is correctly referring you to the JSON-vs-SQL null.

**Reply by ludralph** [reply]
I'm glad you could figure this out 🙌
